src/Parser.zig

```diff
@@ -4772,17 +4802,38 @@ const Result = struct {
     /// Saves value and replaces it with `.unavailable`.
     fn saveValue(res: *Result, p: *Parser) !void {
         assert(!p.in_macro);
-        if (res.val.tag == .unavailable) return;
+        if (res.val.tag == .unavailable or res.val.tag == .nullptr_t) return;
         if (!p.in_macro) try p.value_map.put(res.node, res.val);
         res.val.tag = .unavailable;
     }
 
     fn castType(res: *Result, p: *Parser, to: Type, tok: TokenIndex) !void {
         var cast_kind: Tree.CastKind = undefined;
+
         if (to.is(.void)) {
             // everything can cast to void
             cast_kind = .to_void;
             res.val.tag = .unavailable;
+        } else if (to.is(.nullptr_t)) {
+            if (res.ty.is(.nullptr_t)) {
+                cast_kind = .no_op;
+            } else {
+                try p.errStr(.invalid_object_cast, tok, try p.typePairStrExtra(res.ty, " to ", to));
+                return error.ParsingFailed;
+            }
+        } else if (res.ty.is(.nullptr_t)) {
+            if (to.is(.bool)) {
+                res.ty = to;
+                try res.implicitCast(p, .bitcast);
```

Comment:
It does seem odd but I think it's what clang does https://godbolt.org/z/xj6vcGxjs - the bitcast seemed weird to me because a null pointer isn't necessarily an all-zero representation but I figured maybe that was due to the default target. I can change it to null_to_ptr / ptr_to_bool